libr/util/big.c

```diff
@@ -180,11 +181,11 @@ R_API char *r_big_to_hexstr(RNumBig *b) {
 	sprintf (&ret_str[i], R_BIG_SPRINTF_FORMAT_STR, b->array[j--]);
 	for (; ret_str[i + k] == '0' && k < 2 * R_BIG_WORD_SIZE; k++) {
 	}
-	for (z = k; ret_str[i + z]; z++) {
+	for (z = k; ret_str[i + z] && z < 2 * R_BIG_WORD_SIZE; z++) {
```

Comment:
this 2 * R_BIG_WORD_SIZE its a bit shady, better use a constant defined before with a meaningful name and use it on both loops

Reply:
meaningful name: `last_z` for example